src/RegistrationVerifier.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.30;
+
+import {IDKIMRegistry} from "@zk-email/contracts/interfaces/IDKIMRegistry.sol";
+import {StringUtils} from "@zk-email/contracts/utils/StringUtils.sol";
+import {Verifier} from "../circuits/verifier.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+interface IRegistrationVerifier {
+    enum Intention {
+        REGISTRATION,
+        RECIPIENT_UPDATE
+    }
+
+    function verify(
+        string memory title,
+        address recipient,
+        Intention intention,
+        uint256[2] calldata a,
+        uint256[2][2] calldata b,
+        uint256[2] calldata c,
+        uint256[12] calldata signals
+    ) external view;
+}
+
+contract RegistrationVerifier is IRegistrationVerifier, Ownable {
+    using StringUtils for *;
+
+    string public constant DOMAIN = "gmail.com";
+    string public constant REGISTRATION_PREFIX = "56K66KqN5bey5pS25Yiw5oqV56i/"; // 確認已收到投稿
+    string public constant RECIPIENT_UPDATE_PREFIX = "56K66KqN5q2k5oqV56i/5pu05pS556i/6LK75pS25Y+W5Zyw5Z2A"; // 確認此投稿更改稿費收取地址
+
+    IDKIMRegistry public dkimRegistry;
+    Verifier public verifier;
+    string public sender = "";
```

Comment:
For the final version maybe we can set them as `immutable` but right now we can implement it the way that's most convenient for you.

Reply:
Or even concatenate `sender` and `DOMAIN`

Reply:
> Because we need to verify the domain and the email sender separately, it might not be suitable to concatenate them together.

Got it. Or what if we store the hash of the string and then compare hashes to check if string matches instead? I see the `stringEq` from zkemail's library is used. Wonder if behind the scene it's also using the hashes for string comparison.

Reply:
Yes, behind the scene `stringEq` is using the hash for string comparison. Sure, we can also store the hash of the sender string.